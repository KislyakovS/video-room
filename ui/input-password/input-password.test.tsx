import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import InputPassword from './input-password';

interface MockProps {
    type: string;
}

jest.mock('../icon', () => ({ type }: MockProps) => `Svg Icon - ${type}`);

describe('<InputPassword>', () => {
    it('snapshot', () => {
        const { container } = render(<InputPassword />);

        expect(container).toMatchSnapshot();
    });

    it('it follows that when you click on the buttons, the type `input` should become text', async () => {
        const user = userEvent.setup();

        const { container, getByRole } = render(<InputPassword />);
        const input = container.querySelector('input')!;
        const button = getByRole('button');

        await user.click(button);

        expect(input.type).toBe('text');
    });
});