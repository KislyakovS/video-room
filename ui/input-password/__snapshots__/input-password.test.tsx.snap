// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputPassword> snapshot 1`] = `
<div>
  <div
    class="wrapper"
  >
    <input
      class="input indentRight fullWidth"
      type="password"
    />
    <div
      class="endAdornment"
    >
      <button
        aria-label="Hide password"
        class="button"
        type="button"
      >
        Svg Icon - eyeClose
      </button>
    </div>
  </div>
</div>
`;
